tests/rules/deprecation.test.ts

```diff
@@ -169,6 +169,38 @@ ruleTester.run('deprecation', rule, {
     getValidTestCase(`
       const component = <Component/>;
     `),
+    getValidTestCase(`
+    class Class {
+      method(param: string): void;
+      /** @deprecated */ method(param: number): void;
+
+      method(param: any): void {}
+    }
+    new Class().method('');
+    const obj = new Class();
+    obj.method('');
+  `),
+  getValidTestCase(`
+    class Class {}
+    interface Interface extends Class {
+      method(param: string): void;
+      /** @deprecated */ method(param: number): void;
+    }
+    const obj: Interface = { method(args: any) {} };
+    obj.method('');
+  `),
+  getValidTestCase(`
+    interface Interface {}
+    class Class implements Interface {
+      method(param: string): void;
+      /** @deprecated */ method(param: number): void;
+
+      method(param: any): void {}
+    }
+    new Class().method('');
+    const obj = new Class();
+    obj.method('');
```

Comment:
This test case seems to be exactly the same as the first one with extra interface implementation which does not seem like it adds any value to the test case.
Can we remove this one?